pytensor/xtensor/rewriting/shape.py

```diff
@@ -105,3 +119,73 @@ def lower_transpose(fgraph, node):
     x_tensor_transposed = x_tensor.transpose(perm)
     new_out = xtensor_from_tensor(x_tensor_transposed, dims=out_dims)
     return [new_out]
+
+
+@register_xcanonicalize
+@node_rewriter([ExpandDims])
+def local_expand_dims_reshape(fgraph, node):
+    """Rewrite rule to convert expand_dims to pytensor.tensor.expand_dims and broadcast_to if needed."""
+    if not isinstance(node.op, ExpandDims):
+        return False
+
+    x = node.inputs[0]
+    dim = node.op.dim
+    size = getattr(node.op, "size", 1)
+
+    # If dim is None, don't add a new dimension (matching xarray behavior)
+    if dim is None:
+        return [x]
+
+    # Create new dimensions list with the new dimension at the beginning
+    new_dims = [dim, *list(x.type.dims)]
+
+    # Create new shape with the new dimension at the beginning
+    new_shape = [1, *list(x.type.shape)]
+
+    # Convert to tensor and use pytensor.tensor.expand_dims
+    x_tensor = tensor_from_xtensor(x)
+    x_tensor_expanded = expand_dims(x_tensor, axis=0)
+    if size != 1:
+        x_tensor_expanded = broadcast_to(x_tensor_expanded, new_shape)
+    new_out = xtensor_from_tensor(x_tensor_expanded, dims=tuple(new_dims))
+    return [new_out]
+
+
+@register_xcanonicalize
+@node_rewriter([Squeeze])
+def local_squeeze_reshape(fgraph, node):
+    """Rewrite rule to convert squeeze to pytensor.tensor.squeeze."""
+    if not isinstance(node.op, Squeeze):
+        return False
+
+    x = node.inputs[0]
+    dim = node.op.dim
+
+    # Convert single dimension to iterable for consistent handling
+    dims_to_remove = [dim] if isinstance(dim, str) else dim
+
+    if dims_to_remove is not None:
+        # Validate dimensions exist and have size 1
+        dim_indices = []
+        for d in dims_to_remove:
+            if d not in x.type.dims:
+                return False
+            dim_idx = x.type.dims.index(d)
+            # Only check shape != 1 if the shape is not None (symbolic)
+            if x.type.shape[dim_idx] is not None and x.type.shape[dim_idx] != 1:
+                return False
+            dim_indices.append(dim_idx)
+    else:
+        # Find all dimensions of size 1
+        dim_indices = [i for i, s in enumerate(x.type.shape) if s == 1]
+        if not dim_indices:
+            return False
+
+    # Create new dimensions list
+    new_dims = [d for i, d in enumerate(x.type.dims) if i not in dim_indices]
```

Comment:
Just reuse `node.outputs[0].type.dims` since you already did the work of figuring out the output dims in `make_node`